pkg/storage/s3/s3.go

```diff
@@ -998,3 +999,76 @@ func sharedCredentialsDataFromStaticCreds(accessKey, accessSecret string) []byte
 
 	return buf.Bytes()
 }
+
+// PutStorageTags is for adding/overwriting tags of the S3 bucket
+// which name is obtained using the ID() method.
+func (d *driver) PutStorageTags(tagList map[string]string) error {
+	if len(tagList) == 0 {
+		klog.Info("TagSet is empty, no action taken")
+		return nil
+	}
+
+	svc, err := d.getS3Service()
+	if err != nil {
+		return err
+	}
+
+	tagset := make([]*s3.Tag, 0, len(tagList))
+	for key, value := range tagList {
+		tagset = append(tagset, &s3.Tag{
+			Key:   aws.String(key),
+			Value: aws.String(value),
+		})
+	}
+
+	_, err = svc.PutBucketTaggingWithContext(d.Context, &s3.PutBucketTaggingInput{
+		Bucket: aws.String(d.ID()),
+		Tagging: &s3.Tagging{
+			TagSet: tagset,
+		},
+	})
+	if err != nil {
+		errMsg := ""
+		if aerr, ok := err.(awserr.Error); ok {
+			errMsg = fmt.Sprintf("%s: %s", aerr.Code(), aerr.Error())
+		} else {
+			errMsg = fmt.Sprintf("Unknown Error Occurred: %s", err.Error())
+		}
+		return fmt.Errorf("failed to update s3 bucket tags: %s", errMsg)
+	}
+
+	return nil
+}
+
+// GetStorageTags is fetching the tags of the S3 bucket which name
+// is obtained using the ID() method.
+// If no tags are present(NoSuchTagSet error) is considered as
+// successful scenario.
+func (d *driver) GetStorageTags() (map[string]string, error) {
+	svc, err := d.getS3Service()
+	if err != nil {
+		return nil, err
+	}
+
+	output, err := svc.GetBucketTaggingWithContext(d.Context, &s3.GetBucketTaggingInput{
+		Bucket: aws.String(d.ID()),
+	})
+	if err != nil {
+		errMsg := ""
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() != "NoSuchTagSet" {
+				errMsg = fmt.Sprintf("%s: %s", aerr.Code(), aerr.Error())
+			}
+		} else {
+			errMsg = fmt.Sprintf("Unknown Error Occurred: %s", err.Error())
+		}
+		return nil, fmt.Errorf("failed to fetch s3 bucket tags: %s", errMsg)
+	}
+
+	tagList := make(map[string]string)
+	for _, tags := range output.TagSet {
+		tagList[aws.StringValue(tags.Key)] = aws.StringValue(tags.Value)
+	}
+
+	return tagList, nil
```

Comment:
nit: keep naming consistent throughout this package
```suggestion
	tags := make(map[string]string)
	for _, tag := range output.TagSet {
		tag[aws.StringValue(tag.Key)] = aws.StringValue(tag.Value)
	}

	return tags, nil
```